lib/views/saved_schedule/saved_schedule_layout.dart

```diff
@@ -42,6 +62,94 @@ class _SavedScheduleLayoutState extends State<SavedScheduleLayout> {
 
     Provider.of<ScheduleLayoutSettingProvider>(context, listen: false)
         .initialConditionSubjectTitle(widget.savedSchedule.subjectTitleSetting);
+
+    _courseValidator = CourseValidatorPass(widget.savedSchedule.subjects!.length);
+
+  }
+
+  void _onRefresh() async {
+    
+    /*
+      NOTE:
+      For now, we only use data from the title field of the SavedSchedule object.
+
+      TODO: Save student's kuliyyah into HiveDB
+    */
+
+    // Get kuliyyah code (e.g: KICT) from tokenized title
+    final kuliyyah = name.split(' ').elementAt(0);
+
+    // Keep track of the current subject index
+    var currentIndex = 0;
+
+    print(widget.savedSchedule.subjects);
+
+    // Here, we loop through each of student's saved subject and get the latest data from IIUM's database
+    await Future.forEach<SavedSubject>(widget.savedSchedule.subjects!.toList(), (subject) async {
+
+      // Update state of the pull-to-refresh loading text
+      // (We use course code instead of course name to prevent text overflowing)
+      setState(() => currentRefreshCourse = 'Getting latest data for ${subject.code}');
+
+      final response = await SubjectFetcher.fetchSubjectData(
+        albiruni: Albiruni(
+          semester: widget.savedSchedule.semester,
+          session: widget.savedSchedule.session
+        ),
+        kulliyyah: kuliyyah,
+        courseCode: subject.code,
+        section: subject.sect
+      );
+
+      _courseValidator.subjectSuccess(currentIndex++, response);
+
+    });
+
+    if (!_courseValidator.isClearToProceed()) {
+      ScaffoldMessenger.of(context).showSnackBar(const SnackBar(content: Text('We\'re facing some issues while fetching latest data. Try again later.')));
+      return;
+    }
+
+    /**
+     * Update student's hiveDB SavedSchedule with the latest subject
+     * 
+     * TODO: Refactorize code as a separate method
+     */
+
+    SavedSchedule currentSchedule = widget.savedSchedule;
+    List<Subject> updatedSubjectsList = _courseValidator.fetchedSubjects();
+    
+    currentSchedule.subjects = currentSchedule.subjects!.map((prevSubject) {
+      final updatedSubject = updatedSubjectsList.singleWhere((subject) => (subject.code == prevSubject.code));
+      
+      /**
+       * In some cases, a course venue can be empty in i-Maluum and lecturers will update through
+       * other platforms (i.e: Whatsapp / Microsoft Teams). Since users will override the
+       * venue all by themselves, it is necessary to skip empty venue from i-Maluum
+       * as it will override the current venue.
+       * 
+       * Refer https://github.com/iqfareez/iium_schedule/pull/51#pullrequestreview-1158961053
+       */
+      if (updatedSubject.venue == '') return prevSubject;
```

Comment:
```suggestion
      if (updatedSubject.venue == null) return prevSubject;
```
Checking if the string is empty doesn't guarantee the venue is empty. I think checking for null is better, in-line with the albiruni implementation:

https://github.com/iqfareez/albiruni/blob/b8c561eb33ca71625b231bcb3fab993ed64a8952/lib/src/albiruni_base.dart#L152

Reply:
Fixed the code, thanks for the new knowledge!

Reply:
No problem